src/main/java/jenkins/scm/api/SCMEvent.java

```diff
@@ -247,14 +304,79 @@ public int hashCode() {
     @Override
     public String toString() {
         return String.format(
-                "SCMEvent{type=%s, timestamp=%tc, payload=%s}",
+                "SCMEvent{type=%s, timestamp=%tc, payload=%s, origin=%s}",
                 type,
                 timestamp,
-                payload
+                payload,
+                StringUtils.defaultIfBlank(origin, ORIGIN_UNKNOWN)
         );
     }
 
     /**
+     * Helper method to get the origin of an event from a {@link StaplerRequest}.
+     *
+     * @param req the {@link StaplerRequest}.
+     * @return the origin of the event.
```

Comment:
Be more specific about the format, and/or give an example. And under which conditions might it be null?